cmd/nerdctl/rmi.go

```diff
@@ -78,7 +78,7 @@ func rmiAction(cmd *cobra.Command, args []string) error {
 	walker := &imagewalker.ImageWalker{
 		Client: client,
 		OnFound: func(ctx context.Context, found imagewalker.Found) error {
-			if found.MatchCount > 1 {
+			if found.MatchCount > 1 && !force {
```

Comment:
NOLGTM, It will create ambiguity if have the same short prefix.

Reply:
Q: Is this not what docker does though?
```
➜  nerdctl git:(volume-inspect-size) ✗ docker image rmi 62aedd01bd85 
Error response from daemon: conflict: unable to delete 62aedd01bd85 (must be forced) - image is referenced in multiple repositories
➜  nerdctl git:(volume-inspect-size) ✗ docker image rmi -f 62aedd01bd85
Untagged: abc:latest
Untagged: busybox:latest
Untagged: busybox@sha256:e63286d6bf23acce079c38cd9f63ec19dd1ad4da0e3e942e39f2a08050b50534
Deleted: sha256:62aedd01bd8520c43d06b09f7a0f67ba9720bdc04631a8242c65ea995f3ecac8
Deleted: sha256:7ad00cd55506625f2afad262de6002c8cef20d214b353e51d1025e40e8646e18
```

Reply:
`image is referenced in multiple repositories` mean they have the same image id.

But `rmi 62a` is no guarantee that the same image id is deleted.

Reply:
Agree with @junnplus, we need to guarantee that we delete all images with the same id, not just with the same prefix

Reply:
you should compare the found Digest with args